File: 41/cpp/main.cpp
#include <iostream>

class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int missing_positive = 1;
        std::set<int> num_set(nums.begin(), nums.end());
        for (int num : num_set) {
            if (num == missing_positive) {
                missing_positive++;
            } else if (num > missing_positive) {
                break;
            }
        }
        return missing_positive;
    }
};

int main() {
    Solution sol;

    // Test case X: Y
    RETURN_TYPE resultX = sol.FUNCTION_NAME(Y);
    std::cout << "Test X: " << resultX << std::endl;

    return 0;
}

File: 41/python3/main.py
#!/usr/bin/env python3

class Solution:
    def firstMissingPositive(self, nums):
        # Convert the list to a set for fast lookups
        num_set = set(nums)
        
        missing_positive = 1  # Start with 1 as the smallest positive integer

        # Increment missing_positive until we find a missing number
        while missing_positive in num_set:
            missing_positive += 1

        return missing_positive
